modules/optimization/src/executioners/SteadyAndAdjoint.C

```diff
@@ -23,6 +23,13 @@ SteadyAndAdjoint::validParams()
   params.set<MooseEnum>("solve_type") = "newton";
   params.suppressParameter<MooseEnum>("solve_type");
 
+  // The adjoint system (second one) is solved by _adjoint_solve
+  // This is a parameter of the MultiSystemSolveObject, which we set from here, the executioner.
+  // We seek to prevent the MultiSystemSolveObject from solving both systems
+  // This is abusing input parameters, but SolveObjects do not have their own syntax
+  // and we need to send this parameter from the executioner to the default nested SolveObject
+  params.addPrivateParam<bool>("solve_only_first_system", true);
```

Comment:
could also be cleaner to set the system_names here and not make them providate. I have not made up my mind